### PR TITLE
Add version 10.0 keyboard compiler and development tool support

### PR DESCRIPTION
**Pull request by mcdurdin**
This reworks some of the v10.0 compiler code to ensure that v10.0 keyboards (e.g. web keyboard with chiral support) won't be loaded in KeymanWeb, and makes sure that lower version keyboards compile without v10.0 dependencies.

V10.0 keyboards will require an additional update to Keyman Engine on Windows, Mac, in order to load. This will be handled in a separate PR.

### CODE REVIEW

File: windows/src/developer/kmcmpdll/Compiler.cpp [hunk 1]
```diff
@@ -1254,7 +1254,9 @@ DWORD ProcessSystemStore(PFILE_KEYBOARD fk, DWORD SystemID, PFILE_STORE sp)
 	case TSS_VERSION:
 		if((fk->dwFlags & KF_AUTOMATICVERSION) == 0) return CERR_VersionAlreadyIncluded;
 		p = sp->dpString;
-		if(*p < '5') AddWarning(CWARN_OldVersion);
+    if (wcstof(p, NULL) < 5.0) {
```
Comment (darcywong00) [inline windows/src/developer/kmcmpdll/Compiler.cpp]: Would it be possible to check against `VERSION_MIN`?

File: windows/src/developer/TIKE/compile/CompileKeymanWeb.pas [hunk 2]
```diff
@@ -889,7 +893,30 @@ function TCompileKeymanWeb.JavaScript_Shift(fkp: PFILE_KEY; FMnemonic: Boolean):
...
+          ReportError(fkp.Line, CWARN_ExtendedShiftFlagsNotSupportedInKeymanWeb, 'Extended shift flags LALT, RALT, LCTRL, RCTRL are not supported in KeymanWeb');
+        end;
+
+        if (fkp.ShiftFlags and (
+          KMX_CAPITALFLAG or KMX_NOTCAPITALFLAG or KMX_NUMLOCKFLAG or KMX_NOTNUMLOCKFLAG or KMX_SCROLLFLAG or KMX_NOTSCROLLFLAG)) <> 0 then   // I4118
+        begin
+          ReportError(fkp.Line, CWARN_ExtendedShiftFlagsNotSupportedInKeymanWeb, 'Extended shift flags CAPS and NCAPS are not supported in KeymanWeb');
+        end;
+
+        Result := KMX_ISVIRTUALKEY or (Integer(fkp.ShiftFlags) and $70)
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/compile/CompileKeymanWeb.pas]: Fix this magic number

File: windows/src/developer/kmcmpdll/Compiler.cpp (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/kmcmpdll/Compiler.cpp]: VERSION_MIN is a hexadecimal 0x500, not a value 5.0. We could divide by 0x100 but it's less clear :)

Comment (mcdurdin) [reply windows/src/developer/kmcmpdll/Compiler.cpp]: Note that the reason for this change is that version 10.0 was showing as a lower version than 5.0 with the old code.

**Pull request merged**